pushy/src/main/java/com/turo/pushy/apns/ApnsChannelFactory.java

```diff
@@ -84,7 +85,8 @@
         this.bootstrapTemplate.option(ChannelOption.TCP_NODELAY, true);
         this.bootstrapTemplate.remoteAddress(apnsServerAddress);
         this.bootstrapTemplate.resolver(proxyHandlerFactory == null ?
-                DefaultAddressResolverGroup.INSTANCE : NoopAddressResolverGroup.INSTANCE);
+                new RoundRobinDnsAddressResolverGroup(ClientChannelClassUtil.getDatagramChannelClass(eventLoopGroup),
```

Comment:
For readers/reviewers, this is the key change. Pretty much everything else is just adding dependencies on Netty's DNS resolver module or expanding `ClientChannelClassUtil` to cover datagram sockets, too.